oak/server/rust/oak_runtime/src/runtime/mod.rs

```diff
@@ -72,6 +73,10 @@ struct NodeInfo {
     /// (with one small exception, when the Runtime is in the process of closing
     /// down and the [`JoinHandle`] is held by the shutdown processing code).
     join_handle: Option<JoinHandle<()>>,
+
+    /// If the Node is currently running, holds the [`oneshot::Sender`] for
```

Comment:
Is it worth coalescing `join_handle` and `notify_sender` into a single struct (not sure what name), so that it can be held inside a single Option, and also it may expose a single method to notify and wait at the same time. WDYT?

Reply:
Maybe `struct NodeStopper` or `NodeNotifier`, which one is better?

Reply:
`NodeTerminator` probably will be too much =)

Reply:
`NodeNotifier` seems fine, we can rename later if we come up with something better.

Reply:
The new struct will still need to have `Option` arguments, because [`join`](https://doc.rust-lang.org/std/thread/struct.JoinHandle.html) takes ownership of `self`.

Reply:
I think you can `take` the entire struct from the option, and then I think you can just use its components? But if not, don't worry about it for now